src/preset_cli/cli/superset/sync/dbt/datasets.py

```diff
@@ -36,6 +36,18 @@ def model_in_database(model: ModelSchema, url: SQLAlchemyURL) -> bool:
     return model["database"] == url.database
 
 
+def clean_metadata(metadata: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Remove incompatbile columns from metatada.
+    When updating an existing column/metric we need to remove some fields from the payload.
+    """
+    for key in ("changed_on", "created_on", "type_generic"):
+        if key in metadata:
+            del metadata[key]
+
+    return metadata
```

Comment:
that's a great idea! I think for now the scope might be limited, but I'll check if this is used in other parts of the code (not necessarily related with the dbt sync) and I can improve this in a future PR

Reply:
Unsure why, but the suggested change breaks our current tests. It might require updating the tests as well. I'll try to cover this too while simplifying `sync_datasets`.